components/slider/src/use-slider-styles.ts

```diff
@@ -193,6 +217,64 @@ export const useSliderStyles_unstable = (state: SliderState): SliderState => {
     state.track.className
   );
 
+  const hasSectionLabelsAndColors = sectionLabels.length > 0;
+
+  if (hasSectionLabelsAndColors) {
+    const gradientDirection = !rtl ? "right" : "left";
+
+    const trackColorGradient = useMemo(() => {
```

Comment:
Conditional memos is a no no. Not sure why there isn't a lint warning or something for this, but this needs to be rewritten to have the memo run always.

Reply:
Yeah, I'm used to memos to not be allowed in conditions and also noticed the linter not to catch it. I figured the consolidation outcome can not be changed as long as the range slider state is not changed but I can try to find a better way, and allowed way to di it.